tests/test_ops/test_rans.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import pickle
+import time
+
+import pytest
+import torch
+
+from mmcv.ops import (pmf_to_quantized_cdf, rans_decode_with_indexes,
+                      rans_encode_with_indexes)
+
+EXTEND_TIMES = 100
+DEVICES = ['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']
+
+
+@pytest.fixture(scope='module')
+def rans_input_data():
+    inputs = torch.load('tests/data/for_rans/rans_encoder_inputs.pt')
+    symbols = inputs['symbols'].repeat(EXTEND_TIMES)
+    indexes = inputs['indexes'].repeat(EXTEND_TIMES)
+    cdfs = inputs['quantized_cdfs']
+    cdfs_sizes = inputs['cdfs_sizes']
+    offsets = inputs['offsets']
+    return symbols, indexes, cdfs, cdfs_sizes, offsets
+
+
+@pytest.fixture(scope='module')
+def rans_output_data():
+    with open('tests/data/for_rans/rans_encoder_output.pkl', 'rb') as f:
+        encoded = pickle.load(f)
+    return encoded
+
+
+@pytest.fixture(scope='module')
+def pmf_input_data():
+    inputs = torch.load('tests/data/for_rans/pmf_to_quantized_cdf_inputs.pt')
+    pmfs = inputs['pmfs']
+    pmf_lengths = inputs['pmf_lengths']
+    tail_masses = inputs['tail_masses']
+    return pmfs, pmf_lengths, tail_masses
+
+
+@pytest.fixture(scope='module')
+def pmf_output_data():
+    output = torch.load('tests/data/for_rans/pmf_to_quantized_cdf_output.pt')
+    quantized_cdfs = output['quantized_cdfs']
+    return quantized_cdfs
+
+
+def _move_data_to_target_device(data, device):
+    if isinstance(data, torch.Tensor):
+        return data.to(device)
+    return [d.to(device) for d in data]
+
+
+def _calculate_mibs(fn, device, num_threads, num_symbols, args):
+    if torch.cuda.is_available():
```

Comment:
use torch.cuda.event to do the cuda timing

Reply:
I change it into torch.cuda.event.
```suggestion
    if torch.cuda.is_available():
        start_event = torch.cuda.Event(enable_timing=True)
        end_event = torch.cuda.Event(enable_timing=True)
        start_event.record()
```